### PR TITLE
xcm: Support MultiAssets outbound transfers 

### PR DESCRIPTION
**Pull request by NunoAlexandre**
# Description

With the current `BaseCallFilter` set in all our runtimes, it is _not_ possible to build custom XCM message using `pallet_xcm` or `orml_xtokens` to transfer multiassets all together. We introduced this call filter to block all backdoors that would allow users to transfer Tranche tokens through XCM without the right permission checks. At the time we had no use case needing those calls in the first place, we just blocked them all. 

However, we now have an upcoming use case where to transfer xcUSDC tokens from Centrifuge to Moonbeam we will need either `transfer_multiassets` or `transfer_multiasset_with_fee`.

## Changes and Descriptions

We now have more clarity surrounding how Tranche tokens will be transferred with permissions through Connectors, which won't make use of the common XCM fungibles config. Thus we can simply rely on the fact that `CurrencyIdConvert` doesn't handle Tranche tokens and therefore those extrinsics that do allow to specify Tranche tokens either directly `CurrencyId::Tranche` or as a MultiLocation will fail to execute.

That means that drop the BaseCallFilter and add tests verifying that `CurrencyIdConvert` doesn't handle Tranche tokens and that said extrinsics fail to execute when involving Tranche tokens.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] integration-tests for `restricted_calls` for Altair and Centrifuge
- [x] integration tests covering CurrencyIdConvert handling of Tranche tokens for Altair and Centrifuge

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [ ] I have performed a self-review of my code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I rebased on the latest `main` branch

### CODE REVIEW

**Review by mustermeiszer — COMMENTED** [review]
I have a question regarding failing test in the allowed section. 

On another note: Would `pallet_xcm` also allow to create arbitrary XCM calls? Is this implicitly blocked with the current `CurrencyIdConverted`?

File: runtime/integration-tests/src/xcm/kusama/restricted_calls.rs [hunk 1]
```diff
@@ -41,75 +43,128 @@ use crate::xcm::kusama::{
...
+					]))),
+					0,
+					Box::new(
+						MultiLocation::new(
+							1,
+							X2(
+								Parachain(PARA_ID_SIBLING),
+								Junction::AccountId32 {
+									network: NetworkId::Any,
+									id: BOB.into(),
```
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply runtime/integration-tests/src/xcm/kusama/restricted_calls.rs:158]: > Why is this in the allowed section

@mustermeiszer what "allowed" section are you referring to?

> Shouldnt the X2` junction go through here?

I am also not understanding this one, sorry 🙉  In any case, this MultiLocation / X2 is referring to the destination of this XCM; the tranche token is referenced in the `VersionedMultiAssets::from(MultiAssets::from(vec![tranche_multi_asset])))` block above.

**Comment by NunoAlexandre** [comment]
> On another note: Would `pallet_xcm` also allow to create arbitrary XCM calls? Is this implicitly blocked with the current `CurrencyIdConverted`?


Yes that's right! The reason I didn't write an explicit test for the calls that can be built using `pallet_xcm` is that those are raw messages so a lot more cumbersome to build and they share the same underlying `XcmConfig` and therefore `CurrencyIdConvert` which we rely on to stop Tranche tokens from being XCM transferrable.

File: runtime/integration-tests/src/xcm/kusama/restricted_calls.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply runtime/integration-tests/src/xcm/kusama/restricted_calls.rs:158]: Ah, nevermind, you deleted the `mod allowed` section. Did overread that.

**Review by mustermeiszer — APPROVED** [review]
LGTM!

File: runtime/integration-tests/src/xcm/kusama/restricted_calls.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply runtime/integration-tests/src/xcm/kusama/restricted_calls.rs:158]: Ah ok I see :)

**Review by branan — APPROVED** [review]
Much more straightforward than the old filter, I like it :+1:

**Pull request merged**